File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    n = len(nums)//2
    nums = list(set(nums))
    tmp = []
    for i in nums:
        if i not in tmp:
            tmp.append(i)
    if len(tmp) > n:
        answer = n
    else:
        answer = len(tmp)
    return answer
    

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    const monster = {}
    for (let i of nums) {
        monster[i] = (monster[i] || 0) + 1
    }
    
    const kind = Object.keys(monster).length
    if (kind >= nums.length/2) {
        answer = nums.length/2
    }
    else {
        answer = kind
    }
    
    return answer;
}